src/main/java/com/couchbase/lite/listener/LiteServlet.java

```diff
@@ -182,5 +186,25 @@ public Credentials credentialsWithBasicAuthentication(HttpServletRequest req) {
         return null;
     }
 
+    private URL remoteURL(HttpServletRequest req) {
+        String addr = req.getRemoteAddr();
+        Log.v(Log.TAG_LISTENER, "remoteURL() addr=" + addr);
+        if (!"127.0.0.1".equals(addr) && !"::1".equals(addr)) {
+            if (addr.indexOf(':') >= 0)
+                addr = String.format("[%s]", addr);
+
+            // Java URL can not set username
+            //String username = allowedCredentials != null ? allowedCredentials.getLogin() : null;
```

Comment:
From http://docs.oracle.com/javase/6/docs/api/java/net/URL.html:
set(String protocol, String host, int port, String authority, String userInfo, String path, String query, String ref) 

Is userInfo = username (&password)?